app/src/main/java/com/github/mik629/android/fundamentals/data/repositories/MoviesRepositoryImpl.kt

```diff
@@ -16,14 +18,17 @@ import kotlinx.coroutines.async
 import kotlinx.coroutines.awaitAll
 import kotlinx.coroutines.coroutineScope
 import timber.log.Timber
+import java.util.concurrent.atomic.AtomicBoolean
 import java.util.concurrent.atomic.AtomicReference
 import javax.inject.Inject
 
 class MoviesRepositoryImpl @Inject constructor(
     private val serverApi: ServerApi,
-    private val dao: MovieDao
+    private val dao: MovieDao,
+    private val context: Context
 ) : MoviesRepository {
     private var imageBaseUrl: AtomicReference<String> = AtomicReference()
```

Comment:
imageBaseUrl: AtomicReference<String?> = AtomicReference(null)
isUpdateCacheWorkerRunning: AtomicBoolean = AtomicBoolean(false)

Reply:
Done.
So to have always predicted and self-documented behaviour (+ in case if they decide to change initial value, though highly unlikely)?

Reply:
right
they or you